File: fetch2/src/main/java/com/tonyodev/fetch2/FetchHelper.java
package com.tonyodev.fetch2;

import android.content.Context;

import java.util.IllegalFormatCodePointException;
import java.util.List;

import okhttp3.OkHttpClient;

final class FetchHelper {

    private FetchHelper() {}

    static void throwIfContextIsNull(Context context) {

        if(context == null) {
            throw new IllegalArgumentException("Context cannot be null");
        }
    }

    static String getDefaultDatabaseName(){
        return "com_tonyodev_fetch";
    }


    static void throwIfFetchNameIsNullOrEmpty(String databaseName) {

        if(databaseName == null || databaseName.isEmpty()) {
            throw new IllegalArgumentException("DatabaseManager Name cannot be null or empty");
        }
    }

    static void throwIfGroupIDIsNull(String groupId) {

        if (groupId == null) {
            throw new IllegalArgumentException("groupId cannot be null");
        }
    }

    static void throwIfRequestIsNull(Request request) {

        if(request == null) {
            throw new IllegalArgumentException("Request cannot be null");
        }
    }

    static void throwIfClientIsNull(OkHttpClient client) {

        if(client == null) {
            throw new IllegalArgumentException("OkHttpClient cannot be null");
        }
    }

    static void throwIfStatusIsNull(Status status) {

        if(status == null) {
            throw new IllegalArgumentException("Status cannot be null");
        }
    }

    static void throwIfRequestListIsNull(List<Request> list) {

        if(list == null) {
            throw new IllegalArgumentException("List<Request> cannot be null");
        }
    }

    static void throwIfIdListIsNull(List<Long> list) {

        if(list== null) {
            throw new IllegalArgumentException("List<Long> cannot be null");
        }
    }

    static void throwIfCallbackIsNull(Callback callback) {

        if(callback == null) {
            throw new IllegalArgumentException("Callback cannot be null");
        }
    }

    static void throwIfQueryIsNull(Query query) {

        if(query == null) {
            throw new IllegalArgumentException("Query cannot be null");
        }
    }

    static void throwIfDisposed(Disposable disposable) {

        if(disposable.isDisposed()) {
            throw new DisposedException("This instance cannot be reused after disposed is called");
        }
    }

    static long[] createIdArray(List<Long> ids) {

        for (Long id : ids) {
            if(id == null) {
                throw new NullPointerException("id inside List<Long> cannot be null");
            }
        }

        long[] idArray = new long[ids.size()];
        for (int i = 0; i < ids.size(); i++) {
            idArray[i] = ids.get(i);
        }

        return idArray;
    }
}

File: fetch2/src/main/java/com/tonyodev/fetch2/FetchHelper.kt
package com.tonyodev.fetch2

import android.content.Context

import okhttp3.OkHttpClient

internal object FetchHelper {

    val defaultDatabaseName: String
        get() = "com_tonyodev_fetch"

    fun throwIfContextIsNull(context: Context?) {

        if (context == null) {
            throw IllegalArgumentException("Context cannot be null")
        }
    }


    fun throwIfFetchNameIsNullOrEmpty(databaseName: String?) {

        if (databaseName == null || databaseName.isEmpty()) {
            throw IllegalArgumentException("DatabaseManager Name cannot be null or empty")
        }
    }

    fun throwIfGroupIDIsNull(groupId: String?) {

        if (groupId == null) {
            throw IllegalArgumentException("groupId cannot be null")
        }
    }

    fun throwIfRequestIsNull(request: Request?) {

        if (request == null) {
            throw IllegalArgumentException("Request cannot be null")
        }
    }

    fun throwIfClientIsNull(client: OkHttpClient?) {

        if (client == null) {
            throw IllegalArgumentException("OkHttpClient cannot be null")
        }
    }

    fun throwIfStatusIsNull(status: Status?) {

        if (status == null) {
            throw IllegalArgumentException("Status cannot be null")
        }
    }

    fun throwIfRequestListIsNull(list: List<Request>?) {

        if (list == null) {
            throw IllegalArgumentException("List<Request> cannot be null")
        }
    }

    fun throwIfIdListIsNull(list: List<Long>?) {

        if (list == null) {
            throw IllegalArgumentException("List<Long> cannot be null")
        }
    }

    fun throwIfCallbackIsNull(callback: Callback?) {

        if (callback == null) {
            throw IllegalArgumentException("Callback cannot be null")
        }
    }

    fun throwIfQueryIsNull(query: Query<*>?) {

        if (query == null) {
            throw IllegalArgumentException("Query cannot be null")
        }
    }

    fun throwIfDisposed(disposable: Disposable) {

        if (disposable.isDisposed) {
            throw DisposedException("This instance cannot be reused after disposed is called")
        }
    }

    fun createIdArray(ids: List<Long>): LongArray {

        for (id in ids) {
            if (id == null) {
                throw NullPointerException("id inside List<Long> cannot be null")
            }
        }

        val idArray = LongArray(ids.size)
        for (i in ids.indices) {
            idArray[i] = ids[i]
        }

        return idArray
    }
}
